android/app/src/main/java/org/smartregister/fhircore/activity/QuestionnaireActivity.kt

```diff
@@ -86,13 +93,101 @@ class QuestionnaireActivity : AppCompatActivity() {
 
     val patient = ResourceMapper.extract(questionnaire, questionnaireResponse) as Patient
 
-    patient.id = patient.name.first().family
-    // FhirApplication.fhirEngine(applicationContext).save(patient)
+    patient.id =
+      intent.getStringExtra(PatientDetailFragment.ARG_ITEM_ID) ?: patient.name.first().family
+
     viewModel.savePatient(patient)
 
     this.startActivity(Intent(this, PatientListActivity::class.java))
   }
 
+  private fun getQuestionnaire(): String {
+    val questionnaire =
+      FhirContext.forR4().newJsonParser().parseResource(viewModel.questionnaire) as Questionnaire
+
+    intent.getStringExtra(PatientDetailFragment.ARG_ITEM_ID)?.let {
+      var patient: Patient? = null
+      viewModel.viewModelScope.launch {
+        patient = FhirApplication.fhirEngine(applicationContext).load(Patient::class.java, it)
+      }
+
+      patient?.let {
+
+        // set first name
+        questionnaire.item[0].item[0].item[0].apply {
```

Comment:
how do we know that the first name is the first item in the questionnaire? and so on for below

Reply:
followed the ordering of `patient-registration.json` form items.

Reply:
The ordering in that json enforced somehow or could someone rewrite that to change the order?

Is there a way to implement this using lookups without assuming a particular ordering?

Reply:
The work on [Questionnaire population with another resource](https://github.com/google/android-fhir/issues/273) would cover this since the population of the questionnaire for editing would work regardless of the order of questions.

Reply:
ok, do you think we should merge this as is for now and replace later, or try to find another way now?

Reply:
I have updated the logic to find an item by the `linkId` instead of index. Now if we change the order of items so that will don't affect the behavior.

Reply:
ok cool I think that's less brittle, thanks